hydrolib/core/io/ini/parser.py

```diff
@@ -239,8 +239,26 @@ def _handle_datarow(self, line: str) -> None:
 
     def _retrieve_property_comment(self, line: str) -> Tuple[Optional[str], str]:
         if self._config.parse_comments and self._config.comment_delimiter in line:
-            key_value, comment = line.split(self._config.comment_delimiter, 1)
-            return comment.strip(), key_value.strip()
+            # key_value, comment = line.split(self._config.comment_delimiter, 1)
+            parts = line.strip().split(self._config.comment_delimiter)
+            numhash = len(parts) - 1
+            if numhash == 1:
+                # normal value, simple comment: "key =  somevalue # and a comment "
+                return parts[-1].strip(), parts[0].strip()
+            elif parts[0] == "":
```

Comment:
This logic is odd to read. Might be better to write `numhash == 2` (case without comment).

Reply:
It is deliberate: it catches these cases:
```
key = #valueA# # and a comment
key = #valueA with spaces # # Comment: and give me your #1 reason for writing such a comment?
```
if the first token of split is empty string (and numhas > 1), then we have a starting "#", which must always mean hash-string-quoting.

Reply:
It catches this silly case:
```
key = valueA # and here is hash #2, funny huh?
```

Reply:
See latest commit